src/types/Color.ts

```diff
@@ -0,0 +1,89 @@
+import * as _ from 'lodash'
+
+export const Color: SBVRType<number, RGBA> = {
+	types: {
+		postgres: 'INTEGER',
+		mysql: 'INTEGER',
+		websql: 'INTEGER',
+		odata: {
+			name: 'Self.Color',
+			complexType: `
+				<ComplexType Name="Color">
+					<Property Name="r" Nullable="false" Type="Edm.Int8"/>
+					<Property Name="g" Nullable="false" Type="Edm.Int8"/>
+					<Property Name="b" Nullable="false" Type="Edm.Int8"/>
+					<Property Name="a" Nullable="false" Type="Edm.Int8"/>
+				</ComplexType>
+			`
+			}
+	},
+
+	nativeProperties: {
+		has: {
+			'Red Component': (from:string) => ['BitwiseAnd', ['BitwiseShiftRight', from, 16], 255],
+			'Green Component': (from:string) => ['BitwiseAnd', ['BitwiseShiftRight', from, 8], 255],
+			'Blue Component': (from:string) => ['BitwiseShiftRight', from, 255],
+			'Alpha Component': (from:string) => ['BitwiseAnd', ['BitwiseShiftRight', from, 24], 255],
+		}
+	},
+
+	fetchProcessing: (data, callback) => {
+		callback(null, {
+			r: (data >> 16) & 0xFF,
+			g: (data >> 8) & 0xFF,
+			b: data & 0xFF,
+			a: (data >> 24) & 0xFF,
+		})
+	},
+
+	validate: (value, required, callback) => {
+		let processedValue = 0
+		let err = ''
+		if (!_.isObject(value)) {
+			processedValue = _.parseInt(value)
+
+			if (_.isNaN(processedValue)) {
+				err = `is neither an integer or color object: ${value}`
```

Comment:
This never calls the callback - can you add tests for these cases as well if they're not being picked up by existing tests